oxidation/crates/parsec_core_fs/src/error.rs

```diff
@@ -0,0 +1,66 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use thiserror::Error;
+use uuid::Uuid;
+
+use parsec_api_crypto::CryptoError;
+use parsec_api_types::FileDescriptor;
+
+#[derive(Error, Debug)]
+pub enum FSError {
```

Comment:
Would be useful to add the errors returned by `diesel` to their proper variants.

I don't like to lose the info of the reason of the error return by the library `diesel`